Source/Core/Core/CheatSearch.cpp

```diff
@@ -0,0 +1,327 @@
+      if (!matches_func)
+        continue;
+
+      const auto current_value =
+          ReadValueFromEmulatedMemory(addr, previous_result.m_value.m_type, value_size);
+      if (matches_func(current_value, value ? *value : previous_result.m_value))
+      {
+        results.emplace_back();
+        results.back().m_address = addr;
+        results.back().m_value = current_value;
+      }
+    }
+  });
+
+  return results;
+}
+
+std::optional<std::vector<Cheats::SearchResult>>
+Cheats::NextSearch(const std::vector<Cheats::SearchResult>& previous_results,
+                   Cheats::CompareType comparison)
+{
+  return NextSearchInternal(previous_results, comparison, nullptr);
+}
+
+std::optional<std::vector<Cheats::SearchResult>>
+Cheats::NextSearch(const std::vector<Cheats::SearchResult>& previous_results,
+                   Cheats::CompareType comparison, const Cheats::SearchValue& value)
+{
+  return NextSearchInternal(previous_results, comparison, &value);
+}
+
+std::optional<std::vector<Cheats::SearchResult>>
+Cheats::UpdateValues(const std::vector<Cheats::SearchResult>& previous_results)
+{
+  if (!Memory::m_pRAM)
+    return std::nullopt;
+
+  std::vector<Cheats::SearchResult> results;
+  results.reserve(previous_results.size());
+  Core::RunAsCPUThread([&] {
+    for (const Cheats::SearchResult& previous_result : previous_results)
+    {
+      const u32 addr = previous_result.m_address;
+      if (!PowerPC::HostIsRAMAddress(addr, true))
+        continue;
+
+      const auto current_value = ReadValueFromEmulatedMemory(addr, previous_result.m_value.m_type,
+                                                             GetValueSize(previous_result.m_value));
+      results.emplace_back();
+      results.back().m_address = addr;
+      results.back().m_value = current_value;
+    }
+  });
+
+  return results;
+}
+
+std::vector<u8> Cheats::GetValueAsByteVector(const Cheats::SearchValue& value)
+{
+  std::vector<u8> result;
+  switch (value.m_type)
+  {
+  case Cheats::DataType::U8:
+    result.push_back(value.m_value_u8);
+    break;
+  case Cheats::DataType::U16:
+    result.push_back((value.m_value_u16 >> 8) & 0xFF);
+    result.push_back(value.m_value_u16 & 0xFF);
+    break;
+  case Cheats::DataType::U32:
+    result.push_back((value.m_value_u32 >> 24) & 0xFF);
+    result.push_back((value.m_value_u32 >> 16) & 0xFF);
+    result.push_back((value.m_value_u32 >> 8) & 0xFF);
+    result.push_back(value.m_value_u32 & 0xFF);
+    break;
+  case Cheats::DataType::F32:
+    result.push_back((static_cast<u32>(value.m_value_f32) >> 24) & 0xFF);
+    result.push_back((static_cast<u32>(value.m_value_f32) >> 16) & 0xFF);
+    result.push_back((static_cast<u32>(value.m_value_f32) >> 8) & 0xFF);
+    result.push_back(static_cast<u32>(value.m_value_f32) & 0xFF);
+    break;
+  case Cheats::DataType::F64:
+    result.push_back((static_cast<u64>(value.m_value_f64) >> 56) & 0xFF);
```

Comment:
BitCast is sensible, however I don't think the begin/end construction would be equivalent, as this array is intended to be used as a drop-in memory patch (see `UpdatePatch()` in `CheatsManager.cpp`), so the array has to be constructed in big endian order rather than host endian order. Should probably be noted on the function, too.

Reply:
That can easily be resolved by doing a htobe64 (or equivalent) first.

Reply:
Indeed. That's why I left the assignment as `// ...` for brevity. A `Common::swap64(Common::BitCast<u64>(...))` can take its place.

Reply:
Alright, I made a helper function for the conversion. This looks good to me now.

Reply:
Looks good! The helper functions should be declared static or placed within an anonymous namespace to give it internal linkage though

Reply:
Yup, did that!